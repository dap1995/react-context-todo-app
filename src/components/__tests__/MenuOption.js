import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Option from '../MenuOption';

describe('MenuOption Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Option onClick={() => {}} />, div);
  });

  describe('should be', () => {
    it('enabled', () => {
      const tree = renderer.create(<Option enabled={true} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('disabled', () => {
      const tree = renderer.create(<Option enabled={false} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
