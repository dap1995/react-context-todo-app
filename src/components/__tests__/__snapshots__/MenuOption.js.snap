// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuOption Component should be disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #bfbfbf;
  background-color: white;
  padding: 5px;
  margin: 5px;
  font-size: 12px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: white;
  border: 1px solid transparent;
  border-radius: 3px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: border-color 0.2s linear;
  transition: border-color 0.2s linear;
}

<a
  className="c0"
/>
`;

exports[`MenuOption Component should be enabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #bfbfbf;
  background-color: white;
  padding: 5px;
  margin: 5px;
  font-size: 12px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: white;
  border: 1px solid transparent;
  border-radius: 3px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: border-color 0.2s linear;
  transition: border-color 0.2s linear;
  border-color: rgba(175,47,47,0.5);
}

<a
  className="c0"
/>
`;
